lib/ocpp/v201/charge_point.cpp

```diff
@@ -3603,6 +3616,11 @@ void ChargePoint::cache_cleanup_handler() {
                 EVLOG_debug << "Time based authorization cache cleanup";
             }
             this->auth_cache_cleanup_required = false;
+            lk.unlock();
+        }
```

Comment:
You should also add the `stop_auth_cache_cleanup_handler` to the predicate lambda in the `wait_for`. Currently with the notify it will wake up the thread, run the predicate and see that it does not actually need to wake up so it might sleep for another 15 minutes.